src/main/java/com/redhat/devtools/intellij/quarkus/lsp4ij/LSPEditorWrapper.java

```diff
@@ -0,0 +1,246 @@
+import com.intellij.codeInsight.daemon.impl.DaemonCodeAnalyzerImpl;
+import com.intellij.codeInsight.daemon.impl.HighlightInfo;
+import com.intellij.openapi.application.ApplicationManager;
+import com.intellij.openapi.editor.Document;
+import com.intellij.openapi.editor.Editor;
+import com.intellij.openapi.editor.RangeMarker;
+import com.intellij.openapi.editor.event.*;
+import com.intellij.openapi.editor.ex.EditorEx;
+import com.intellij.openapi.project.Project;
+import com.intellij.openapi.util.Key;
+import com.intellij.openapi.util.Pair;
+import com.intellij.openapi.util.TextRange;
+import com.intellij.openapi.vfs.VirtualFile;
+import com.intellij.psi.PsiDocumentManager;
+import com.intellij.psi.PsiFile;
+import com.intellij.util.containers.ContainerUtil;
+import com.redhat.devtools.intellij.quarkus.lsp4ij.operations.codeactions.LSPCodeActionIntentionAction;
+import com.redhat.devtools.intellij.quarkus.lsp4ij.operations.codeactions.LSPCodeActionResult;
+import org.eclipse.lsp4j.*;
+import org.eclipse.lsp4j.jsonrpc.messages.Either;
+import org.eclipse.lsp4j.services.LanguageServer;
+import org.jetbrains.annotations.NotNull;
+
+import java.net.URI;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
+
+public class LSPEditorWrapper {
+
+    private static final Key<LSPEditorWrapper> KEY = new Key(LSPEditorWrapper.class.getName());
+
+    private final Editor editor;
+
+    private final EditorMouseMotionListener editorMouseMotionListener;
+
+    private final CaretListener caretListener;
+
+    private final DocumentListener documentListener;
+
+    private boolean documentChanged;
+
+    private LSPCodeActionResult codeActionResult;
+
+    private LSPEditorWrapper(Editor editor) {
+        this.editor = editor;
+        this.editorMouseMotionListener = new EditorMouseMotionListener() {
+            @Override
+            public void mouseMoved(@NotNull EditorMouseEvent e) {
+                Document document = editor.getDocument();
+                VirtualFile file = LSPIJUtils.getFile(document);
+                List<Diagnostic> currentDiagnostics = LSPVirtualFileWrapper.getLSPVirtualFileWrapper(file).getAllDiagnostics();
+                if (currentDiagnostics != null && !currentDiagnostics.isEmpty()) {
+                    int offset = getTargetOffset(e);
+                    if (offset != -1) {
+                        for (Diagnostic diagnostic : currentDiagnostics) {
+                            int startOffset = LSPIJUtils.toOffset(diagnostic.getRange().getStart(), document);
+                            int endOffset = LSPIJUtils.toOffset(diagnostic.getRange().getEnd(), document);
+                            if (offset >= startOffset && offset <= endOffset) {
+                                try {
+                                    updateQuickFixAt(offset,startOffset,endOffset).get();

```

Comment:
`updateQuickFixAt` can return `null`. Should `null`-checked before calling `get()` on the `CompletableFuture`